### PR TITLE
Add support for --(enable|disable)-nls to configure.

### PR DESCRIPTION
**Pull request by illiminable**
- Allows disabling of Native Language Support (NLS).
- Default is enabled to maintain current behavior.

### CODE REVIEW

**Comment by illiminable** [comment]
It allows a platform to not build/install data that will never be used. For example Chrome OS globally disables the nls USE flag in portage in order to save space since the user can never interact with such tools. Chrome OS also forbids packages to write to /usr/shared/locale to prevent this in the case a package doesn't support -NLS use flag.

eg. This change was he start of all packages not being allowed to install this data - https://codereview.chromium.org/6284004/

An additional change will also go to gentoo/portage to wire the USE flag (other packages that support this - https://packages.gentoo.org/useflags/nls ) up to this configure option.

File: src/configure.in [hunk 1]
```diff
@@ -118,15 +118,34 @@ LIBUTIL=-lutil
...
-			[Define if translation functions should be used.])])])
-
-AC_CHECK_PROG(MSGFMT,msgfmt,msgfmt)
+# Determine if NLS is desired and supported.
 po=
-if test x"$MSGFMT" != x; then
-	po=po
+AC_ARG_ENABLE([nls],
+AC_HELP_STRING([--disable-nls],
+               [Disable Native Language Support(NLS).]), explicit=yes,
```
Comment (vapier) [inline src/configure.in]: please indent with tabs instead of spaces

**Comment by greghudson** [comment]
Sorry, I let this fall through the cracks over the holidays.

I prefer using a single tristate variable (as in the aesni support) to two separate flags.  In particular, note that if this pattern were repeated for two different settings, an explicit --with option for the first one would also apply to the second.

**Comment by vapier** [comment]
sorry, but i don't understand what you're asking for here.  `--{en,dis}able-nls` flags are standard in the open source autoconf world.

**Comment by illiminable** [comment]
I didn't get back to this after the holidays yet. I will update and follow the aesni pattern. And address the other formatting issues.

**Comment by vapier** [comment]
sure, that makes sense.  autoconf autocreates `$enable_nls` already, and the use of `$enableval` outside of the `AC_ARG_ENABLE` macro is generally a bad idea.

File: src/configure.in (hunk 1 above)
Comment (illiminable) [inline src/configure.in]: This file has a mix of tabs and space but the majority of the file is indented like this with spaces. All the AC_HELP_STRING that are multi-line are like this (except 2 with have one tab and then spaces).

Seems better to stick with the existing style unless there's a real compelling reason.

Comment (illiminable) [inline src/configure.in]: The other more common pattern in here seems to be 2 spaces for indentation.

Comment (greghudson) [inline src/configure.in]: Don't worry about the indentation; as you noted, configure.in is wildly internally inconsistent with its indentation style.  If I see a need to change the formatting of the code, I can do that during the merge.

**Pull request merged**